core/src/main/java/com/pingcap/tikv/datatype/TypeMapping.java

```diff
@@ -90,6 +80,9 @@ public static org.apache.spark.sql.types.DataType toSparkType(DataType type) {
       if (type.isUnsigned() && type.getType() == TypeLonglong) {
         return DataTypes.createDecimalType(20, 0);
       }
+      if (type.getType() == TypeTiny && type.getLength() == 1) {
```

Comment:
Hey @TrafalgarRicardoLu It appeals this is a breaking change, can explain more why this change is required?

Reply:
I think no. IIRC MySQL doesn't have boolean type before, so everyone use 1 bit tinyint to represent a boolean type. Many MySQL ecosystem softwares still streat this as bool.

Reply:
Yea, understand this is trying to align with others databases, but for someone who rely on this before, after upgrading to the new version, they possibly meet issues. Just an advice, maybe we can communicate such changes in the changelog to alert users who relied on the previous behavior.

Reply:
Yeah, reasonable. We could do better. Does this change affect you?

Reply:
Yea, we actually use this to ingest data from TiDB to lakehouse through HUDI, and because of this schema change, the target table could meet schema incompatible issues.